src/Tasks/Common/Resources/Strings.resx

```diff
@@ -588,4 +588,47 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1120: C++/CLI projects targeting .NET Core require a target framework of at least 'netcoreapp3.1'.</value>
     <comment>{StrBegin="NETSDK1120: "}</comment>
   </data>
+  <data name="AppHostMachONotExpectedFormat" xml:space="preserve">
+    <value>NETSDK1121: Unable to use '{0}' as macOS application host executable it was not in the expected format. Detail reason: {1}.</value>
```

Comment:
@wli3 If we catch this exception, it almost always means that the apphost executable in the package is corrupt. 
So, I think the customer facing main error should say something like: $"Application host {apphost} is unusable; the file is likely corrupted. Please reinstall the dotnet SDK version {version}"

Reply:
@nguerrera @KathleenDollard what do you think? If there is no use for the detail error it's less to maintain.